src/Symfony/Component/VarDumper/Server/Connection.php

```diff
@@ -9,104 +9,88 @@
  * file that was distributed with this source code.
  */
 
-namespace Symfony\Component\VarDumper\Dumper;
+namespace Symfony\Component\VarDumper\Server;
 
 use Symfony\Component\VarDumper\Cloner\Data;
-use Symfony\Component\VarDumper\Dumper\ContextProvider\ContextProviderInterface;
+use Symfony\Component\VarDumper\Server\ContextProvider\ContextProviderInterface;
 
 /**
- * ServerDumper forwards serialized Data clones to a server.
+ * Forwards serialized Data clones to a server.
  *
  * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
  */
-class ServerDumper implements DataDumperInterface
+class Connection
 {
     private $host;
-    private $wrappedDumper;
     private $contextProviders;
     private $socket;
 
     /**
      * @param string                     $host             The server host
-     * @param DataDumperInterface|null   $wrappedDumper    A wrapped instance used whenever we failed contacting the server
      * @param ContextProviderInterface[] $contextProviders Context providers indexed by context name
      */
-    public function __construct(string $host, DataDumperInterface $wrappedDumper = null, array $contextProviders = array())
+    public function __construct(string $host, array $contextProviders = array())
     {
         if (false === strpos($host, '://')) {
             $host = 'tcp://'.$host;
         }
 
         $this->host = $host;
-        $this->wrappedDumper = $wrappedDumper;
         $this->contextProviders = $contextProviders;
+        $this->socket = $this->createSocket();
```

Comment:
Let's connect early. That's not an issue since this is async and it lets some time for the connection to be opened (might be useful: this is async)